### ISSUE TITLE
Protractor ReactJS automation code is failed with this error "Failed: javascript error: Timed out"

### ISSUE THREAD

**Issue by bharadwaj92-qa** [abhinaba1080/protractor-react-selector#8]
Steps:
1. I have launched the ReactJS application in the Chrome browser and then maximized the window screen.
2. Now my requirement is, my code should be able to click on "Sign up" button in that React application.
**Website is**: 'https://www.codecademy.com/
**Code used:**
it('Able to identify the react elements', async () => {
   await browser.waitForReact(8000,'#ReduxBinder'); // ReduxBinder is the root element in the application
//Below Buttonbase is component and signup is a prop
   const signButton=element(by.react('ButtonBase',{children:"Sign up"}));
    await signButton.click();
});

Result: In browser application is launched and immediately after 3-4 secs, browser is closed, code is failed and showed the error as:
**"Failed: javascript error: Timed out"**

**Comment by bharadwaj92-qa** [comment]
Closing the issue because I have found the reason.
Reason: As element is NOT identified when I have used "By.react" method, so it threw me timed out error. When I have attempted with "By.xpath" method and I have passed the prop value in it, so now code is able to interact with that sign up button.
**Code used**:
const signing=element(by.xpath("//*[@data-testid='header-sign-up']"));
//data-testid is the prop name
signing.click();

**Comment by bharadwaj92-qa** [comment]
Closing the issue because I have found the reason.
Reason: As element is NOT identified when I have used "By.react" method, so it threw me timed out error. When I have attempted with "By.xpath" method and I have passed the prop value in it, so now code is able to interact with that sign up button.
**Code used**:
const signing=element(by.xpath("//*[@data-testid='header-sign-up']"));
//data-testid is the prop name
signing.click();
